File: swift/Sources/UtmParse/UtmParams.swift
import Foundation

/// UTM Parameters container
public struct UtmParams {
    public let source: String?
    public let medium: String?
    public let campaign: String?
    public let term: String?
    public let content: String?
    public let platform: String?

    public init(
        source: String? = nil,
        medium: String? = nil,
        campaign: String? = nil,
        term: String? = nil,
        content: String? = nil,
        platform: String? = nil
    ) {
        self.source = source
        self.medium = medium
        self.campaign = campaign
        self.term = term
        self.content = content
        self.platform = platform
    }

    public var isEmpty: Bool {
        source == nil && medium == nil && campaign == nil && term == nil && content == nil
    }
}

File: java/utm-parse/src/main/java/com/tanstackship/utm/UtmParams.java
package com.tanstackship.utm;

/**
 * UTM Parameters container
 */
public class UtmParams {
    private String source;
    private String medium;
    private String campaign;
    private String term;
    private String content;
    private String platform;

    public UtmParams() {}

    public String getSource() { return source; }
    public void setSource(String source) { this.source = source; }

    public String getMedium() { return medium; }
    public void setMedium(String medium) { this.medium = medium; }

    public String getCampaign() { return campaign; }
    public void setCampaign(String campaign) { this.campaign = campaign; }

    public String getTerm() { return term; }
    public void setTerm(String term) { this.term = term; }

    public String getContent() { return content; }
    public void setContent(String content) { this.content = content; }

    public String getPlatform() { return platform; }
    public void setPlatform(String platform) { this.platform = platform; }

    public boolean isEmpty() {
        return source == null && medium == null && campaign == null 
            && term == null && content == null;
    }

    @Override
    public String toString() {
        return "UtmParams{source=" + source + ", medium=" + medium + 
               ", campaign=" + campaign + ", term=" + term + 
               ", content=" + content + "}";
    }
}
